kmir/src/kmir/prove.py

```diff
@@ -1,8 +1,8 @@
 import logging
 from pathlib import Path
-from typing import Any, Final, Iterable
+from typing import Any, Final, Iterable, Optional
```

Comment:
We use `_ | None` instead of `Optional[_]` throughout all our codebases.

Reply:
KMIR uses `Optional` throughout the code base. 
Is there any implementation issues that requires `_ | None` than `Optional`?

Reply:
We are trying to use consistent codestyle everywhere, so we should open a separate issue to make this codebase consistent with our codestyle elsewhere. @tothtamas28 I thought that `Optional[X]` vs `X | None` was supposed to be caught by our linters, is this repository not setup correctly with our python project template? https://github.com/runtimeverification/python-project-template

Please make an issue and link it here to get the codestyle updated.

Reply:
> is this repository not setup correctly with our python project template?

It is, it just needs to be updated (`cruft update`), and the `make pyupgrade` step added to CI.

Reply:
@tothtamas28 We are not aware of these options. The project was created by you if I remember correctly.
But it will be great if there could be a documentation about what should be done to correctly set up the project. 
Do you have one? Or maybe you can write an updated one so far?

Meanwhile, would you please submit an issue in this repo regarding the desired settings?
This will be of great help to the new team @sskeirik.

Reply:
Thank you, @tothtamas28. 
Now is it safe to resolve this conversation? @ehildenb